### PR TITLE
Add additional metadata to spooling tracing

### PR DESCRIPTION
**Pull request by wendigo**
## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: core/trino-spi/src/main/java/io/trino/spi/protocol/SpooledSegmentHandle.java [hunk 1]
```diff
@@ -29,4 +29,14 @@ public interface SpooledSegmentHandle
     String encoding();
 
     Instant expirationTime();
+
+    default long rows()
+    {
+        return -1;
```
Comment (losipiuk) [inline core/trino-spi/src/main/java/io/trino/spi/protocol/SpooledSegmentHandle.java:37]: Did you read K&R ANSI C recently?

Comment (losipiuk) [reply core/trino-spi/src/main/java/io/trino/spi/protocol/SpooledSegmentHandle.java:37]: It does not seem that metrics should be part of handle. Let's chat about it.

Comment (wendigo) [reply core/trino-spi/src/main/java/io/trino/spi/protocol/SpooledSegmentHandle.java:37]: why not?

Comment (wendigo) [reply core/trino-spi/src/main/java/io/trino/spi/protocol/SpooledSegmentHandle.java]: Ok, I will leave this information on the `create` span where we have this information in the `SpoolingContext`

File: core/trino-spi/src/main/java/io/trino/spi/protocol/SpooledSegmentHandle.java [hunk 2]
```diff
@@ -13,13 +13,15 @@
...
 
+import io.trino.spi.Experimental;
 import io.trino.spi.QueryId;
 
 import java.time.Instant;
 
 /**
  * SpooledSegmentHandle is used to uniquely identify a spooled segment and to manage its lifecycle.
  */
+@Experimental(eta = "2025-05-31")
```
Comment (losipiuk) [inline core/trino-spi/src/main/java/io/trino/spi/protocol/SpooledSegmentHandle.java:24]: Optimistic 🍆

**Review by losipiuk — APPROVED** [review]

File: core/trino-spi/src/main/java/io/trino/spi/protocol/SpooledSegmentHandle.java (hunk 2 above)
Comment (wendigo) [reply core/trino-spi/src/main/java/io/trino/spi/protocol/SpooledSegmentHandle.java:24]: Day after my birthday :P It's deadline

**Pull request merged**